packages/front-end/components/Features/Variation.tsx

```diff
@@ -0,0 +1,239 @@
+import React, { forwardRef } from "react";
+import { useSortable } from "@dnd-kit/sortable";
+import { CSS } from "@dnd-kit/utilities";
+import { FaArrowsAlt } from "react-icons/fa";
+import { ExperimentValue, FeatureValueType } from "@/../back-end/types/feature";
+import clsx from "clsx";
+import {
+  decimalToPercent,
+  distributeWeights,
+  percentToDecimal,
+  rebalance,
+} from "@/services/utils";
+import {
+  getVariationColor,
+  getVariationDefaultName,
+} from "@/services/features";
+import MoreMenu from "../Dropdown/MoreMenu";
+import Field from "../Forms/Field";
+import FeatureValueField from "./FeatureValueField";
+import styles from "./VariationsInput.module.scss";
+
+interface SortableProps {
+  i: number;
+  id: string;
+  variation: ExperimentValue & { id?: string };
+  variations: ExperimentValue[] & { id?: string };
```

Comment:
A new union type of `ExperimentValue` with an additional  `id` field - This is good. Could be better if you defined elsewhere in the file - `DraggableExperimentValue`, and exported for use across the two components

The type definition for `variations` looks incorrect (unless I'm mistaken) - It's saying that the array itself has an `id` property but it seems like it should read that each element has an `id` prop ... Taking the `DraggableExperimentValue` approach, you should be able to define correctly

```
  variation: DraggableExperimentValue;
  variations: DraggableExperimentValue[];
```